src/WebJobs.Script.WebHost/Controllers/InstanceController.cs

```diff
@@ -35,11 +36,35 @@ public InstanceController(IEnvironment environment, IInstanceManager instanceMan
         [HttpPost]
         [Route("admin/instance/assign")]
         [Authorize(Policy = PolicyNames.AdminAuthLevel)]
-        public async Task<IActionResult> Assign([FromBody] EncryptedHostAssignmentContext encryptedAssignmentContext)
+        public async Task<IActionResult> Assign([FromBody] HostAssignmentRequest hostAssignmentContext)
         {
-            _logger.LogDebug($"Starting container assignment for host : {Request?.Host}. ContextLength is: {encryptedAssignmentContext.EncryptedContext?.Length}");
+            if (string.IsNullOrEmpty(hostAssignmentContext.EncryptedContext) &&
+                hostAssignmentContext.AssignmentContext is null)
+            {
+                return BadRequest("At least one of 'assignmentContext' or 'encryptedContext' must be provided.");
```

Comment:
I think you should verify that one and only one of these is specified. E.g. it'd also be invalid if BOTH are specified. So you should verify this, and error message when violated would be like "Exactly one of {nameof(HostAssignmentContext.EncryptedContext)} or {nameof(HostAssignmentContext.AssignmentContext)}" must be specified."